packages/@aws-cdk/aws-iot-actions/lib/republish.ts

```diff
@@ -0,0 +1,59 @@
+import * as iam from '@aws-cdk/aws-iam';
+import * as iot from '@aws-cdk/aws-iot';
+import { singletonTopicRuleRole } from './util';
+
+/**
+ * Construction properties for a Republish action.
+ */
+export interface RepublishProps {
+  /**
+   * The Quality of Service (Qos) level to use when republishing messages.
+   *
+   * @default - 0
+   */
+  readonly qos?: number;
+  /**
+   *
+   * The MQTT topic to which to republish the message.
+   *
+   * To republish to a reserved topic, which begins with `$`, use `$$` instead.
+   *
+   * For example, to republish to the device shadow topic
+   * `$aws/things/MyThing/shadow/update`, specify the topic as
+   * `$$aws/things/MyThing/shadow/update`.
+   */
+  readonly topic: string;
+  /**
+   * The IAM role that grants access.
+   *
+   * @default - a role will be created
+   */
+  readonly role?: iam.IRole;
+}
+
+/**
+ * Publishes to a IoT Topic
+ */
+export class Republish implements iot.ITopicRuleAction {
+  constructor(private readonly props: RepublishProps) {
+  }
+
+  public bind(rule: iot.ITopicRule): iot.TopicRuleActionConfig {
+    // Allow rule to publish to topic
+    const role = this.props.role || singletonTopicRuleRole(rule, [new iam.PolicyStatement({
```

Comment:
borrowed from [aws-event-target](https://github.com/aws/aws-cdk/blob/165a3d877b7ab23f29e42e1e74ee7c5cb35b7f24/packages/%40aws-cdk/aws-events-targets/lib/util.ts#L17) and have clear gaps in understanding or the role mechanism in this case.